src/prog/prog_ProgItem.cpp

```diff
@@ -43,7 +43,7 @@ ProgItem::~ProgItem(void) {
  */
 ProgItem *ProgItem::next(void) const {
 	ProgItem *result = (ProgItem *)inhstruct::DLNode::next(); 
-	if(!result || result->atEnd())
+	if(!result || (result && result->atEnd()))
```

Comment:
`(!A || (A && B))` simplifies to `(!A || B)`, so what you added changes nothing. C++ uses lazy evaluation, so if `!result`, then `result->atEnd()` will not be evaluated.

Reply:
Ooooh I see. Let's close this PR then and focus on https://github.com/statinf-otawa/elm/pull/1.